variantlib/commands/generate_index_json.py

```diff
@@ -126,13 +100,39 @@ def generate_index_json(args: list[str]) -> None:
                 )
                 continue
 
-            if vdesc.hexdigest not in known_variants:
-                known_variants[variant_hash] = vdesc
+            if (vhash := vdesc.hexdigest) not in known_variants:
+                known_variants[vhash] = vdesc
+
+            # ============== Variant Providers Processing ================ #
+            if not (
+                wheel_providers := wheel_metadata.get_all(
+                    METADATA_VARIANT_PROVIDER_HEADER, []
+                )
+            ):
+                logger.info(
+                    "%(wheel)s: did not declare any `%(key)s`",
+                    {"wheel": wheel, "key": METADATA_VARIANT_PROVIDER_HEADER},
+                )
+
+            for wheel_provider in wheel_providers:
+                ns, _, provider = wheel_provider.partition(", ")
+                ns = ns.strip()
+                provider = provider.strip()
+                if not ns or not provider:
+                    raise ValueError(
```

Comment:
The crash here is inconsistent with graceful handling of invalid metadata above.

Reply:
Managed gracefully. You're right a complete error is not a good idea if we can avoid